crates/bootstrap/src/lib.rs

```diff
@@ -27,14 +28,201 @@ pub struct ExampleState {
+    fn resumed(&mut self, event_loop: &ActiveEventLoop) {
+        let window = event_loop.create_window(self.attributes.clone()).unwrap();
+        window.set_ime_allowed(true);
+
+        let sample_count = get_sample_count();
+
+        let mut app = pollster::block_on(yakui_app::Graphics::new(&window, sample_count));
+
+        // By default, yakui_winit will measure the system's scale factor and pass
+        // it to yakui.
+        //
+        // Sometimes, it might be desirable to scale the UI by a different factor,
+        // like if your game has a "UI scale" option, if you're writing tests, or
+        // you want to ensure your widgets work at a different scale.
+        //
+        // In these examples, setting the YAKUI_FORCE_SCALE environment variable to
+        // a number will override the automatic scaling.
+        if let Some(scale) = get_scale_override() {
+            app.window_mut().set_automatic_scale_factor(false);
+            self.yak.set_scale_factor(scale);
+        }
+
+        // In these examples, set YAKUI_INSET to force the UI to be contained within
+        // a sub-viewport with the given edge inset on all sides.
+        let inset = get_inset_override();
+        if inset.is_some() {
+            app.window_mut().set_automatic_viewport(false);
+        }
+
+        // Preload some textures for the examples to use.
+        self.monkey = Some(
+            self.yak
+                .add_texture(load_texture(MONKEY_PNG, TextureFilter::Linear)),
+        );
+        self.monkey_blurred = Some(
+            self.yak
+                .add_texture(load_texture(MONKEY_BLURRED_PNG, TextureFilter::Linear)),
+        );
+        self.brown_inlay = Some(
+            self.yak
+                .add_texture(load_texture(BROWN_INLAY_PNG, TextureFilter::Nearest)),
+        );
+        self.custom = Some(app.renderer.add_texture(
+            custom_texture::generate(&app.device, &app.queue),
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+            wgpu::FilterMode::Nearest,
+        ));
+
+        // Add a custom font for some of the examples.
+        let fonts = self.yak.dom().get_global_or_init(Fonts::default);
+        let font = Font::from_bytes(
+            include_bytes!("../assets/Hack-Regular.ttf").as_slice(),
+            FontSettings::default(),
+        )
+        .unwrap();
+
+        fonts.add(font, Some("monospace"));
+
+        self.app = Some(app);
+        self.window = Some(window);
+    }
+
+    fn window_event(
+        &mut self,
+        event_loop: &ActiveEventLoop,
+        window_id: WindowId,
+        event: WindowEvent,
+    ) {
+        // `unwrap` is fine, the window will always be available when
+        // receiving a window event.
+
+        if self.app.as_mut().unwrap().handle_event(
+            &mut self.yak,
+            &Event::<()>::WindowEvent {
+                window_id,
+                event: event.clone(),
+            },
```

Comment:
Instead of having the user wrap their events in WindowEvent/DeviceEvent, can we implement conversions for them in `yakui-winit`? That way, the user doesn't need to clone their `event` value nor write the funny user data turbofish.

Reply:
Is the current version OK?

Reply:
I submitted a new comment elsewhere to continue this train of thought